scripts/linkchecker.py

```diff
@@ -410,10 +411,33 @@ def validate_links(page):
 
     matches = regex.findall(content)
     records = []
+    target_records = []
     for m in matches:
         r = check_target(page, m[0], m[1])
         if r:
             records.append(r)
+            target_records.append(m[1])
+
+    # if multiple records are the same they need not be checked repeatedly
+    # remove paths that are not relative too
+    target_records = set([item for item in target_records
```

Comment:
```suggestion
    target_records = {item for item in target_records
```
Nit: you can put an iterator inside curly brackets to make a [`set comprehension`](https://docs.python.org/3/reference/expressions.html#set-displays) directly.